src/main/scala/cromwell/binding/types/WdlArrayType.scala

```diff
@@ -20,6 +21,15 @@ case class WdlArrayType(memberType: WdlType) extends WdlType {
       WdlArray(WdlArrayType(WdlFileType), wdlArray.value.map(str => WdlFile(str.asInstanceOf[WdlString].value)).toList)
     case wdlArray: WdlArray if wdlArray.wdlType.memberType == memberType => wdlArray
     case wdlArray: WdlArray if wdlArray.wdlType.memberType == WdlAnyType => coerceIterable(wdlArray.value)
+    case wdlArray: WdlArray if wdlArray.wdlType.memberType.isInstanceOf[WdlArrayType] && memberType.isInstanceOf[WdlArrayType] =>
+      val coercedElements = wdlArray.value.map(memberType.coerceRawValue)
+      val coercedArray = coercedElements.collect({case x: Failure[_] => x}) match {
+        case failures if failures.nonEmpty =>
+          throw new UnsupportedOperationException(failures.map(f => s"Could not coerce $wdlArray: ${f.exception.getMessage}").mkString("\n"))
+        case _ => coercedElements.map(_.get) // .get because they're all Success
+      }
+      WdlArray(WdlArrayType(memberType), coercedArray)
```

Comment:
Not sure what's going on between this and #248 but there's obviously a lot of the same changes showing up in both places